src/Draco.Compiler/Internal/Documentation/Extractors/XmlDocumentationExtractor.cs

```diff
@@ -0,0 +1,134 @@
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Xml;
+using Draco.Compiler.Internal.Symbols;
+using Draco.Compiler.Internal.Symbols.Metadata;
+using Draco.Compiler.Internal.Symbols.Synthetized;
+
+namespace Draco.Compiler.Internal.Documentation.Extractors;
+
+/// <summary>
+/// Extracts XML into <see cref="SymbolDocumentation"/>.
+/// </summary>
+internal sealed class XmlDocumentationExtractor
+{
+    public string Xml { get; }
+    public Symbol ContainingSymbol { get; }
+    public MetadataAssemblySymbol Assembly => this.ContainingSymbol.AncestorChain.OfType<MetadataAssemblySymbol>().First();
+
+    public XmlDocumentationExtractor(string xml, Symbol containingSymbol)
+    {
+        this.Xml = xml;
+        this.ContainingSymbol = containingSymbol;
+    }
+
+    /// <summary>
+    /// Extracts the <see cref="Xml"/>.
+    /// </summary>
+    /// <returns>The extracted XMl as <see cref="SymbolDocumentation"/>.</returns>
+    public SymbolDocumentation Extract()
+    {
+        // TODO: exception
+        //       para
+        //       list
+        //       c
+        //       see - not cref links
+        //       seealso
+        //       b ?
+        //       i ?
+
+        var xml = $"""
+            <documentation>
+              {this.Xml}
+            </documentation>
+            """;
+        var doc = new XmlDocument();
+        doc.LoadXml(xml);
+        var node = doc.DocumentElement!.FirstChild;
+        var sections = ImmutableArray.CreateBuilder<DocumentationSection>();
+        var nextNode = null as XmlNode;
+        for (; node is not null; node = nextNode)
+            sections.Add(this.ExtractSection(node, out nextNode));
```

Comment:
Since next node is not referenced after the loop, why not inline it in the for initializer?